src/xopen/__init__.py

```diff
@@ -300,11 +300,15 @@ def __init__(
         # The program may crash due to a non-existing file, internal error etc.
         # In that case we need to check. However the 'time-to-crash' differs
         # between programs. Some crash faster than others.
-        # Therefore we peek the first character(s) of stdout. Peek will block
-        # until it gets at least the amount of characters specified or EOF.
-        # This way we ensure the program has at least decompressed some output,
-        # or stopped before we continue.
-        self.process.stdout.peek(1)  # type: ignore  # stdout is io.BufferedReader if set to PIPE.
+        # Therefore we peek the first character(s) of stdout. Peek will return at
+        # least one byte of data, unless at EOF in which case we should wait for the
```

Comment:
I think this sentence is a bit misleading because the docs say that `peek()` actually returns more or fewer bytes than requested https://docs.python.org/3.8/library/io.html#io.BufferedReader.peek . Because of that, I also wonder whether it is possible that the loop that you added below doesn’t terminate (that is, if the process keeps running but `peek()` keeps returning empty output).

Reply:
Ah, see this comment in the implementation of `BZ2File.peek` https://github.com/python/cpython/blob/bb3e0c240bc60fe08d332ff5955d54197f79751c/Lib/bz2.py#L145-L155:
> Relies on the undocumented fact that BufferedReader.peek() always returns at least one byte (except at EOF), independent of the value of n

Reply:
As far as I understood from this thread: https://bugs.python.org/issue5811 Seek should return at least one byte unless at EOF, but the documentation does not mention this.